WcaOnRails/app/models/registration.rb

```diff
@@ -86,6 +83,10 @@ def best_solve(event, type)
     person ? person.best_solve(event, type) : SolveTime.new(event.id, type, 0)
   end
 
+  def event_picker_events
+    registration_events.map(&:event_object).sort_by(&:rank)
+  end
```

Comment:
@jonatanklosko I guess creating an events method might work but I worry that it might impact performance and also most people would expect events to be querying the events table not competition_events. I don't have a better answer at the moment though.